wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -569,23 +572,66 @@ private RecipeParser getRecipeParser(StageContext context)
 
   @Override
   public Relation transform(RelationalTranformContext relationalTranformContext, Relation relation) {
-    if (PRECONDITION_LANGUAGE_SQL.equalsIgnoreCase(config.getPreconditionLanguage())
-            && checkPreconditionNotEmpty(true)) {
+    if (!(PRECONDITION_LANGUAGE_SQL.equalsIgnoreCase(config.getPreconditionLanguage())
+            && checkPreconditionNotEmpty(true))) {
+      return new InvalidRelation("Plugin is not configured for relational transformation");
+    }
 
-      if (!Feature.WRANGLER_PRECONDITION_SQL.isEnabled(relationalTranformContext)) {
-        throw new RuntimeException("SQL Precondition feature is not available");
-      }
+    if (!Feature.WRANGLER_PRECONDITION_SQL.isEnabled(relationalTranformContext)) {
+      throw new RuntimeException("SQL Precondition feature is not available");
+    }
 
-      Optional<ExpressionFactory<String>> expressionFactory = getExpressionFactory(relationalTranformContext);
-      if (!expressionFactory.isPresent()) {
-        return new InvalidRelation("Cannot find an Expression Factory");
-      }
+    Optional<ExpressionFactory<String>> expressionFactory = getExpressionFactory(relationalTranformContext);
+    if (!expressionFactory.isPresent()) {
+      return new InvalidRelation("Cannot find an Expression Factory");
+    }
+
+    Expression filterExpression = expressionFactory.get().compile(config.getPreconditionSQL());
+    Relation filteredRelation = relation.filter(filterExpression);
+
+    ExpressionFactory<String> expFactory = expressionFactory.get();
+
+    String recipe = config.getDirectives();
+
+    registry = SystemDirectiveRegistry.INSTANCE;
+    try {
+      registry.reload("default");
+    } catch (DirectiveLoadException e) {
+      throw new RuntimeException(e);
+    }
 
-      Expression filterExpression = expressionFactory.get().compile(config.getPreconditionSQL());
-      return relation.filter(filterExpression);
+    List<Directive> directives = null;
+    try {
+      GrammarBasedParser parser = new GrammarBasedParser("default", new MigrateToV2(recipe).migrate(), registry);
+      directives = parser.parse();
+    } catch (DirectiveParseException e) {
+      throw new RuntimeException(e);
+    } catch (RecipeException e) {
+      throw new RuntimeException(e);
+    }
+
+    for (Directive directive : directives) {
+      // Expression exp = expFactory.compile(sql);
+      if (!(directive instanceof RelationalDirective)) {
+        throw new RuntimeException("Directive is not relational Directive");
+      }
+      // currently supporting only drop column
+      // SQL will be returned as "DROP COLUMN col1, col2"
+      String sql = ((RelationalDirective) directive).getSQL();
+      List<String> cols = getColumnsOfDropSQL(sql);
+      for (String col : cols) {
+        filteredRelation = filteredRelation.dropColumn(col);
```

Comment:
We should move this logic into the directive. SparkSQLDataset should be returned from each Directive and passed onto the next one. SparkSQLEngine.transform should be invoked in each directive for executing. 
Building just one SQL for each Directive should be good enough. SparkSQLEngine will internally chain the SQL execute in one map operation is my understanding, please check that part. 
There is a comment on that already, https://github.com/cdapio/cdap/blob/1cc26e664a22977fb39f29ea03a46ee4ab531f92/cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/SparkSQLEngine.java#L193

Reply:
Agreed. These changes are for poc. will move the logic to directives.